applications/GeoMechanicsApplication/custom_utilities/transport_equation_utilities.hpp

```diff
@@ -0,0 +1,53 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Gennady Markelov
+//
+
+#pragma once
+
+// Project includes
+
+// Application includes
+#include "geo_mechanics_application_variables.h"
+
+namespace Kratos
+{
+
+class GeoTransportEquationUtilities
+{
+public:
+    template <unsigned int TDim, unsigned int TNumNodes>
+    static inline BoundedMatrix<double, TNumNodes, TNumNodes> CalculatePermeabilityMatrix(
+        const Matrix&                            rGradNpT,
+        double                                   DynamicViscosityInverse,
+        const BoundedMatrix<double, TDim, TDim>& rMaterialPermeabilityMatrix,
+        double                                   RelativePermeability,
+        double                                   PermeabilityUpdateFactor,
+        double                                   IntegrationCoefficient)
+    {
+        return CalculatePermeabilityMatrix(rGradNpT, DynamicViscosityInverse,
+                                           rMaterialPermeabilityMatrix, RelativePermeability,
+                                           PermeabilityUpdateFactor, IntegrationCoefficient);
+    }
+
+    static inline Matrix CalculatePermeabilityMatrix(const Matrix& rGradNpT,
+                                                     double        DynamicViscosityInverse,
+                                                     const Matrix& rMaterialPermeabilityMatrix,
+                                                     double        RelativePermeability,
+                                                     double        PermeabilityUpdateFactor,
+                                                     double        IntegrationCoefficient)
+    {
+        return -PORE_PRESSURE_SIGN_FACTOR * DynamicViscosityInverse *
+               prod(rGradNpT, Matrix(prod(rMaterialPermeabilityMatrix, trans(rGradNpT)))) *
+               RelativePermeability * PermeabilityUpdateFactor * IntegrationCoefficient;
+    }
```

Comment:
Since this is not a template function, I think we can create a definition and implementation (and move the implementation to a .cpp file).